src/main/java/org/dplava/git/RepositoryCommitValidator.java

```diff
@@ -221,29 +220,31 @@ public void run() {
                     LOGGER.debug("Validated changed XML files (" + count + ") since last valid commit in " + timeSince(start) + ".");
                 }
 
+                //make sure all files have unique IDs
                 if (errors.isValid()) {
                     start = System.currentTimeMillis();
                     checkIdentifiers(new HashMap<String, String>(), null, gitDir, errors);
                     LOGGER.debug("Checked XML files for duplicate IDs in " + timeSince(start) + ".");
                 }
                 
                 if (errors.isValid()) {
-                    registry.reportCommitValid(repositoryUri, commitHash);
+                    registry.reportCommitValid(payload);
 
                 } else {
-                    final String reportUrl = reports.writeFailureReport(repositoryUri, commitHash, errors.getErrors());
-                    registry.reportCommitInvalid(repositoryUri, commitHash, reportUrl);
+                    final String reportUrl = reports.writeFailureReport(payload, errors.getErrors());
```

Comment:
Just so that we never leave a commit in the "pending" validation state, could you catch the possible exceptions thrown by this line and before rethrowing them invoke registry.reportCommitInvalid() with a null URL (you may need to update that method to support it).